nemo/collections/common/parts/multi_layer_perceptron.py

```diff
@@ -52,5 +57,6 @@ def forward(self, hidden_states):
             output_states = getattr(self, f'layer{i}')(output_states)
 
         if self.log_softmax:
+            # TODO: make it work with float16? check here
```

Comment:
Please remove the comment

Reply:
Does log_softmax works with float16? I think the reason they did that is that log_softmax was not working properly with float16. Kept it there so that we can check it in our training test. I can remove it if you are OK.

Reply:
@VahidooX then cast to ft32 should be added back (although in most examples data is coming as fl32) if fl16 doesn't work - but why won't it work?

Reply:
I would test it when testing O1 mode. It could be a problem because of the logarithmic operation which need higher precision. I am not sure if it has a problem, but as it was here before, it is good to check it out. If it worked, we can ignore the casting.

Reply:
please remove comment